File: app/src/main/java/me/myshows/android/utils/DateUtils.java
package me.myshows.android.utils;

import android.support.annotation.Nullable;
import android.text.TextUtils;
import android.util.Log;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

/**
 * Created by warrior on 13.09.15.
 */
public class DateUtils {

    private static final String TAG = DateUtils.class.getSimpleName();

    private static final SimpleDateFormat FORMATTER = new SimpleDateFormat("dd.MM.yyyy", Locale.getDefault());
    private static final SimpleDateFormat FORMATTER_ISO8601 = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ", Locale.getDefault());

    @Nullable
    public static Date parse(@Nullable String rawDate) {
        if (!TextUtils.isEmpty(rawDate)) {
            try {
                return FORMATTER.parse(rawDate);
            } catch (ParseException e) {
                Log.e(TAG, e.getMessage(), e);
            }
        }
        return null;
    }

    public static long parseInMillis(@Nullable String rawDate) {
        Date date = parse(rawDate);
        return date != null ? date.getTime() : Long.MAX_VALUE;
    }

    @Nullable
    public static Date parseISO8601(@Nullable String isoString) {
        if (!TextUtils.isEmpty(isoString)) {
            try {
                return FORMATTER_ISO8601.parse(isoString);
            } catch (ParseException e) {
                Log.e(TAG, e.getMessage(), e);
            }
        }
        return null;
    }

    public static long parseInMillisISO8601(@Nullable String isoString) {
        Date date = parseISO8601(isoString);
        return date != null ? date.getTime() : Long.MAX_VALUE;
    }
}

File: app/src/main/kotlin/me/myshows/android/utils/DateUtils.kt
package me.myshows.android.utils

import java.text.SimpleDateFormat
import java.util.*

private val FORMATTER = SimpleDateFormat("dd.MM.yyyy", Locale.getDefault())
private val FORMATTER_ISO8601 = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ", Locale.getDefault())

fun parse(rawDate: String?): Date? =
        when (rawDate?.isEmpty() == false) {
            true -> FORMATTER.parse(rawDate)
            else -> null
        }

fun parseInMillis(rawDate: String?): Long = parse(rawDate)?.time ?: Long.MAX_VALUE

fun parseISO8601(isoString: String?): Date? =
        when (isoString?.isEmpty() == false) {
            true -> FORMATTER_ISO8601.parse(isoString)
            else -> null
        }

fun parseInMillisISO8601(isoString: String?): Long = parseISO8601(isoString)?.time
        ?: java.lang.Long.MAX_VALUE
